### ISSUE TITLE
Self-update fails due to missing 'iaas'-flag

### ISSUE THREAD

**Issue by jkbschmid** [EngineerBetter/control-tower#8]
Cheers!
The `self-update`-job in the `control-tower-self-update`-pipeline fails due to a missing `iaas` flag:
```
+ cd control-tower-release
+ chmod +x control-tower-linux-amd64
+ ./control-tower-linux-amd64 deploy concourse-test
Error validating args on deploy: [failed to validate Deploy flags: [--iaas flag not set]]
```
Concourse was deployed via
`./control-tower-darwin-amd64 deploy  --github-auth-client-id <client-id>  --github-auth-client-secret <client-secret>  --iaas AWS --domain <domain> --region eu-west-1 concourse-test`. 
Version: 0.3.1

Seems like the [aws_pipeline](https://github.com/EngineerBetter/control-tower/blob/7ef8de1fbd35718c207b848d89e0853a6a69cdfe/fly/aws_pipeline.go#L73) (and the gcp_pipeline as well) is missing a `IAAS` parameter. 
I was also wondering whether you could tell me if the `deploy` command in the `self-update`-job is set up so that it retains configuration from the previous deployment (e.g. github-auth-client-id). Thanks!

**Comment by crsimmons** [comment]
Hi @jkbschmid,

Sorry for the delayed response. `control-tower` pulls all flags from the config apart from `iaas`, `region`, and `namespace` (which are used to find the config). So there is no need to provide any other flags on subsequent deploys unless you want to alter one of them.

I've pushed a fix for the self-update pipeline but we currently have no-one on the bench to verify it and cut a release. It has passed our system tests though so if you're feeling adventurous you could try building locally using 68d1d53 and deploying with that.

**Comment by crsimmons** [comment]
@jkbschmid This has been released as part of [0.4.0](https://github.com/EngineerBetter/control-tower/releases/tag/0.4.0). Since the problem was with self-update you will need to manually deploy with this release to update.

Closing this issue - let us know if you have any problems.
